### DISCUSSION TITLE
Welcome to space-nerds-in-space Discussions!

### DISCUSSION THREAD

**Discussion by smcameron** [smcameron/space-nerds-in-space#290] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others. Remember that this is a community we build together.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

* I'm going to turn this github discussion thing on for SNIS and see how it goes.  I expect it will mostly go... silently... but we'll see.  If it turns out to be garbage, I'll turn it off.    Probably if you want discussions on SNIS, you should go over to bridgesim.net or freegamedev.net though.  But, people are reluctant to start an account on some obscure message board, and if you have a github account already, then this will work too at lower friction. -- steve

**Comment by Shadow8472** [comment]
350 days later, I may as well be first response.

Good Morning! I am Shadow_8472 and I've been an on and off fan of the project for several years. I've technically compiled SNIS once but that was just installing the AUR package. Manjaro/KDE's package manager were doing the heavy work. I honestly wasn't expecting that kind of easy mode.

One dream for my blog is a series where I set up a SNIS-powered bridge of my own. A possible cost-saving/portability-enhancing idea I've recently come across but have no idea about its feasibility is using some form of multi-seat configuration where compatible pairs of stations share a tower and have separate user sessions. It might even be fun to get a full bridge running off a single CPU, though I don't see that happening without a specialized rig with multiple graphics cards.

In all seriousness though, a lightweight distro with just enough Linux to play SNIS would increase the game's accessibility a lot. Windows users could fire up a VirtualBox session and go. This hypothetical distro would boot into a menu for users who aren't too sure about the command line. One of those options will be an update/version change script to ease with building from GitHub (or a personal mirror). A hardware benchmark test can recommend graphics settings. Idea inspired by MineOS, a similar concept for home Minecraft servers (now defunct).

**Comment by smcameron** [comment]
I don't know about the separate user sessions on the same machine.  How would the keyboard work?  So far as I know if you plug say two usb keyboards in a machine, they aren't really distinguishable, i.e. you can type on either and both will input into the same window.  And by session do you mean like X11 sessions? so you run an X server on a less powerful machine and use X11's "DISPLAY" variable to redirect a 2nd session to this less powerful X server?  I don't think that works because I don't think GPU accelleration will work to a remote X11 session very well... or maybe it can ( https://evpo.wordpress.com/2017/03/04/opengl-hardware-acceleration-through-remote-x11-ssh-connection/ ) but, if the remote machine is powerful enough to use it's own GPU successfully via X11, then why not just run snis_client on it directly?

As for cheap setup...  Some stations can run alright on a Raspberry Pi 4, (science, comms, engineering, damage control), though it's still not super cheap, you still need monitor, mouse, keyboard, some sort of amplifier and speakers for sound.  Here's a long boring video of SNIS running on a Raspberry Pi 4B: https://www.youtube.com/watch?v=wdy4ICZqc68  Note, had to turn resolution down to 720p to get adequate performance.  Cost is probably around $100-$150 for each station (minus the monitor).  Decent monitor will be around $200 to $300, so probably about $400 per station.  The key to reducing costs here is finding cheap used monitors that will work with the Raspbery Pi 4B.  Googling around, maybe you could use a cheap 24-inch TV as a monitor (they seem to be about $100).

There have been some attempts to make images that you can boot from USB flash drives to enable people to use their windows laptops by booting linux from the USB flash drive.  There is this thing, for instance: https://github.com/fgerling/snis-builder  Note, I have not tried that myself, so caveat emptor.

Finally, it might be worth looking into WSLg on Windows 11.  I have seen SNIS run on Windows 10 WSL, but it used a software renderer and could not access the GPU, so performance wasn't good (maybe still good enough for COMMS, Engineering, or Damage control -- don't really remember).  Purportedly WSLg allows native linux apps to run on Windows and access the GPU, but I have never tried it or seen it tried.

**Reply by Shadow8472** [reply]
In all honesty, the SNIS in a box would be a "because I [think I] can" level of stunt. I recently saw this video: [https://youtu.be/v8tjA8VyfvU
](https://youtu.be/v8tjA8VyfvU) where the presenter covered a product that let XP have up to 6 distinct, extra workstations as demonstrated on a Pentium 4 without any sort of virtualization. Part way in, he drops a note on the screen about Linux having a similar "multi-seat" feature, but warned that it was more fiddly. I researched multi-seat a little already, but I've been more focused on another project lately. I learned that multi-seat was superseded by something with a less memorable name and support for dropped around Debian Buster. I could be a bit hazy on the details. 

This talk of Raspberry Pi's is getting me thinking about how I might arrange a super-deluxe all-in-one/two-in-one simulator with lower-power stations configured as touch screens embedded into a DIY console. Might be a fun configuration to play, especially if you're running an escort scenario with a shuttle pod and a larger ship -- be it a freighter, a damaged ship, or scouting a minefield or other hazardous region. I had a friend go do that when we played Artemis around ten years ago.

I personally have given up entirely on Windows. As noted above, I compiled a version of SNIS from an AUR, but that's about it. Even after a few years daily driving Linux, I still haven't mastered compiling things on Debian systems.
